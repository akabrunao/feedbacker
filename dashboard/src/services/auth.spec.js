import mockAxios from 'axios'
import AuthService from './auth'

jest.mock('axios')

describe('AuthServices', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('deve retornar um token wuando usuÃ¡rio faz login', async () => {
    const token = '123.123.123'
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: { token } })
    })

    const response = await AuthService(mockAxios).login({ email: 'teste@teste.com', password: '1234' })
    expect(response.data).toHaveProperty('token')
    expect(response).toMatchSnapshot()
  })

  it('deve retornar um usuario ao criar um cadastro', async () => {
    const user = {
      name: 'Bruno',
      password: '1234',
      email: 'bruno@bruno.com'
    }
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: user })
    })

    const response = await AuthService(mockAxios).register(user)
    expect(response.data).toHaveProperty('name')
    expect(response.data).toHaveProperty('password')
    expect(response.data).toHaveProperty('email')
    expect(response).toMatchSnapshot()
  })

  it('deve apresentar erro 404', async () => {
    const errors = { status: 404, statusText: 'Not Found' }
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ request: errors })
    })

    const response = await AuthService(mockAxios).login({ email: 'teste@teste.com', password: '1234' })
    expect(response.errors).toHaveProperty('status')
    expect(response.errors).toHaveProperty('statusText')
  })
})
