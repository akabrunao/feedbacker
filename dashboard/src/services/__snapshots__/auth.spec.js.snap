// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthServices deve retornar um token wuando usu√°rio faz login 1`] = `
Object {
  "data": Object {
    "token": "123.123.123",
  },
  "errors": null,
}
`;

exports[`AuthServices deve retornar um usuario ao criar um cadastro 1`] = `
Object {
  "data": Object {
    "email": "bruno@bruno.com",
    "name": "Bruno",
    "password": "1234",
  },
  "errors": null,
}
`;
